kernel/aster-nix/src/syscall/write.rs

```diff
@@ -20,10 +20,6 @@ pub fn sys_write(fd: FileDesc, user_buf_ptr: Vaddr, user_buf_len: usize) -> Resu
         file_table.get_file(fd)?.clone()
     };
 
-    if user_buf_len == 0 {
-        return Ok(SyscallReturn::Return(0));
-    }
-
```

Comment:
There is a TODO in `pwrite64`/`pwritev`:
https://github.com/asterinas/asterinas/blob/52ce2dd5d1c2f1c4ebb5dbe50175c99d2ed09f23/kernel/aster-nix/src/syscall/pwrite64.rs#L24-L27
https://github.com/asterinas/asterinas/blob/52ce2dd5d1c2f1c4ebb5dbe50175c99d2ed09f23/kernel/aster-nix/src/syscall/pwritev.rs#L60-L63

The case for `pwritev` is extremely tricky: If the user provides an empty `IoVec`, we have no way of knowing if the file descriptor is valid for writing, because we never call `FileLike::write` in that case:
```rust
    for io_vec in io_vecs.as_ref() {
        // [..]
        let write_len = file.write(&buffer)?;
        total_len += write_len;
    }
```
I think this is a separate issue, and just like `Socket` should only provide `sendmsg/recvmsg`, perhaps `FileLike` should only provide something like `preadv`/`pwritev`, i.e. the most powerful interface for reading and writing, so that the `FileLike` implementors can report errors like the method is not allowed or the given offset is not supported.

However, this is beyond the scope of this PR.

Reply:
OK, since the writev/readv methods are not added on `FileLike` trait, we can leave it as future work.